docs/tutorials/gluon/mnist.md

```diff
@@ -1,333 +1,434 @@
-We also imported `mxnet.ndarray` package to use activation functions from `ndarray` API.
+In this case, it is rather obvious that our MLP network is either too simple or has not been trained long enough to perform really great on this dataset, as can be seen from the fact that some of the mislabeled examples are rather "easy" and should not be a challenge for our neural net.
+As it turns out, moving to the CNN architecture presented in the following section will give a big performance boost.
 
-Now, We will create the network as follows:
+### Convolutional Neural Network
 
-```python
-net = Net()
-```
+A fundamental issue with the MLP network is that it requires the inputs to be flattened (in the non-batch axes) before they can be processed by the dense layers.
+This means in particular that the spatial structure of an image is largely discarded, and that the values describing it are just treated as a long vector.
+The network then has to figure out the neighborhood relations of pixels from scratch by adjusting its weights accordingly, which seems very wasteful.
 
-![png](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/image/conv_mnist.png)
+A convolutional neural network (CNN) aims to address this problem by using a more structured weight representation.
+Instead of connecting all inputs to all outputs, the characteristic [convolution layer](https://mxnet.incubator.apache.org/api/python/gluon/nn.html#mxnet.gluon.nn.Conv2D) only considers a small neighborhood of a pixel to compute the value of the corresponding output pixel.
+In particular, the spatial structure of the image is preserved, i.e., one can speak of input and output pixels in the first place.
+Only the size of the image may change through convolutions.
+[This article](http://deeplearning.net/software/theano/tutorial/conv_arithmetic.html) gives a good and intuitive explanation of convolutions in the context of deep learning.
 
-**Figure 3:** First conv + pooling layer in LeNet.
+The size of the neighborhood that a convolution layer considers for each pixel is usually referred to as *filter size* or *kernel size*.
+The array of weights -- which does not depend on the output pixel location, only on the position within such a neighborhood -- is called *filter* or *kernel*.
+Typical filter sizes range from *3 x 3* to *13 x 13*, which implies that a convolution layer has *far* fewer parameters than a dense layer.
 
-Now we train LeNet with similar hyper-parameters as before. Note that, if a GPU is available, we recommend using it. This greatly speeds up computation given that LeNet is more complex and compute-intensive than the previous multilayer perceptron. To do so, we only need to change `mx.cpu()` to `mx.gpu()` and MXNet takes care of the rest. Just like before, we'll stop training after 10 epochs.
+```python
+conv_layer = nn.Conv2D(kernel_size=(3, 3), channels=32, in_channels=16, activation='relu')
+print(conv_layer.params)
+# Output:
```

Comment:
Please use the following notation, and remove the commented lines:


     ```python
     conv_layer = nn.Conv2D(kernel_size=(3, 3), channels=32, in_channels=16, activation='relu')
     print(conv_layer.params)
     ```

     `Parameter conv0_weight (shape=(32, 16, 3, 3), dtype=<class 'numpy.float32'>)` <!--notebook-skip-line-->

     `Parameter conv0_bias (shape=(32,), dtype=<class 'numpy.float32'>)` <!--notebook-skip-line-->